### DISCUSSION TITLE
Canvas library issue on deploying to vercel

### DISCUSSION THREAD

**Discussion by kashyapshiv** [vercel/community#903] [Help]
I am deploying my Next JS project on vercel, but I keep getting error of build failing due to canvas library. build is generated on my local machine,

**Comment by swarnava** [comment]
Hi there,

You can refer to https://github.com/orgs/vercel/discussions/884#discussioncomment-3718240



> You have a couple of options that I can think of at the moment.
> 
> One is to move to to CI or deploy from local. You still wouldn't be
> able to keep the Canvas module in the production deployment, but you
> could use it during build and delete as some of the linked solutions
> suggest. See "[How can I use GitHub Actions with
> Vercel?](https://vercel.com/guides/how-can-i-use-github-actions-with-vercel)"
> for more info on how to set up the foundation for that approach.
> 
> Another option is to move the large package into a separate project
> for on-demand use. Some people have done this with things like
> [og-image](https://github.com/vercel/og-image) and [customized
> versions](https://github.com/orgs/vercel/discussions/586#discussioncomment-3181726)
> of it.

**Reply by kashyapshiv** [reply]
Thanks @swarnava , I build via `vercel build` in CLI and serve with `vercel deploy --prebuilt`. The main issue is the size of uncompressed canvas library which is 172MB.

**Reply by swarnava** [reply]
If you run `vercel deploy --prebuilt --archive=tgz`, does it help a bit?
